src/garage/sampler/sampler.py

```diff
@@ -0,0 +1,91 @@
+"""Base sampler class."""
+
+import abc
+import copy
+
+
+class Sampler(abc.ABC):
+    """Abstract base class of all samplers.
+
+    Implementations of this class should override `construct`,
+    `obtain_samples`, and `shutdown_worker`. `construct` takes a
+    `WorkerFactory`, which implements most of the RL-specific functionality a
+    `Sampler` needs. Specifically, it specifies how to construct `Worker`s,
+    which know how to perform rollouts and update both agents and environments.
+
+    Currently, `__init__` is also part of the interface, but calling it is
+    deprecated. `start_worker` is also deprecated, and does not need to be
+    implemented.
+    """
+
+    def __init__(self, algo, env):
+        """Construct a Sampler from an Algorithm.
+
+        Calling this method is deprecated.
+        """
+        self.algo = algo
+        self.env = env
+
+    @classmethod
+    def construct(cls, worker_factory, agents, envs):
```

Comment:
consider naming this `from_worker_factory`, which is a convention for non-default constructors in python. I suppose is could be `from_worker_factor_agents_and_envs` but that seems a little excessive.